testing/conftest.py

```diff
@@ -5,58 +5,78 @@
 import pytest
 from selenium import webdriver
 from selenium.webdriver.common.desired_capabilities import DesiredCapabilities
-from wait_for import wait_for_decorator
+from wait_for import wait_for
 from widgetastic.browser import Browser
 
 
+CAPABILITIES = {
+    "firefox": DesiredCapabilities.FIREFOX,
+    "chrome": {
+        **DesiredCapabilities.CHROME,
+        "chromeOptions": {"args": ["disable-dev-shm-usage", "no-sandbox"]},
+    },
+}
+
+
+def pytest_addoption(parser):
+    parser.addoption(
+        "--browser-name",
+        help="name of the browser",
+        choices=("firefox", "chrome"),
+        default="firefox",
+    )
+
+    parser.addoption("--force-host", default=None, help="foce a selenium hostname")
+
+
 @pytest.fixture(scope="session")
-def browser_name():
-    return os.environ["BROWSER"]
+def browser_name(pytestconfig):
+    return os.environ.get("BROWSER") or pytestconfig.getoption("--browser-name")
 
 
 @pytest.fixture(scope="session")
-def selenium_host(worker_id):
-    oktet = 1 if worker_id == "master" else int(worker_id.lstrip("gw")) + 1
-    host = f"127.0.0.{oktet}"
-    ps = subprocess.run(
-        [
-            "sudo",
-            "podman",
-            "run",
-            "--rm",
-            "-d",
-            "-p",
-            f"{host}:4444:4444",
-            "-p",
-            f"{host}:5999:5999",
-            "--shm-size=2g",
-            "quay.io/redhatqe/selenium-standalone:latest",
-        ],
-        stdout=subprocess.PIPE,
-    )
-    yield host
-    container_id = ps.stdout.decode("utf-8").strip()
-    subprocess.run(["sudo", "podman", "kill", container_id])
+def selenium_url(pytestconfig, worker_id):
+    forced_host = pytestconfig.getoption("--force-host")
+
+    if forced_host is None:
+        oktet = 1 if worker_id == "master" else int(worker_id.lstrip("gw")) + 1
+        host = f"127.0.0.{oktet}"
+        ps = subprocess.run(
+            [
+                "sudo",
+                "podman",
+                "run",
+                "--rm",
+                "-d",
+                "-p",
+                f"{host}:4444:4444",
+                "-p",
+                f"{host}:5999:5999",
+                "--shm-size=2g",
+                "quay.io/redhatqe/selenium-standalone:latest",
+            ],
+            stdout=subprocess.PIPE,
+        )
+
+        print(ps.stdout)
+        yield f"http://{host}:4444/wd/hub"
+
+        container_id = ps.stdout.decode("utf-8").strip()
+        subprocess.run(["sudo", "podman", "kill", container_id])
+    else:
+        yield f"http://{forced_host}:4444/wd/hub"
 
 
 @pytest.fixture(scope="session")
-def wait_for_selenium(selenium_host):
-    @wait_for_decorator(timeout=10, handle_exception=True)
-    def make_request():
-        urlopen(f"http://{selenium_host}:4444/wd/hub")
+def wait_for_selenium(selenium_url):
+    wait_for(lambda: urlopen(selenium_url), timeout=10, handle_exceptions=True)
```

Comment:
That's a root cause of why the tests fail:

```python
wait_for(lambda: urlopen(selenium_url), timeout=10, handle_exception=True)
```

I wanted to make in this as well but I guess made the same mistake. I mixed up `handle_exception` with `handle_exceptions`.

Reply:
Yikes thanks

Reply:
fixed, thanks